packages/mail/src/classes/mail-service.js

```diff
@@ -40,6 +41,17 @@ class MailService {
     return this;
   }
 
+  /**
+   * Reuse connections
+   */
+  setKeepAlive(keepAlive) {
+    if (keepAlive === false) {
+      return;
+    }
+
+    this.client.setDefaultRequest('agent', new Agent({ keepAlive: true }));
```

Comment:
I think the key needs to be `httpsAgent`, according to the [documentation](https://github.com/axios/axios#request-config). This logic should be moved to the underlying `Client` with a separate helper method that we can call from here, similar to calling [`this.client.setApiKey`](https://github.com/khshah6/sendgrid-nodejs/blob/khshah6-add_keep_alive_support/packages/client/src/classes/client.js#L38) from [`setApiKey`](https://github.com/sendgrid/sendgrid-nodejs/pull/1226/files#diff-a4d6dd09858240682deb6dc3cf67bdda6a4b3b5e041537940d56a5f42b7c9a51R38).

Reply:
sure, good call out